Libs/GObject-2.0/Native/Classes/RecordWrapper.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using GLib;
+
+namespace GObject.Native
+{
+    public class RecordWrapper
+    {
+        public static object WrapHandle(IntPtr handle, Type gtype)
+        {
+            System.Type trueType = TypeDictionary.GetSystemType(gtype);
+            
+            if (handle == IntPtr.Zero)
+                throw new NullReferenceException($"Failed to wrap handle as type <{trueType}>. Null handle passed to WrapHandle.");
+
+            // Get constructor for the true type
+            MethodInfo? factory = GetRecordFactory(trueType);
+
+            if (factory == null)
+                throw new Exception($"Type {trueType} does not define an IntPtr constructor. This could mean improperly defined bindings");
+
+            object? result = factory.Invoke(null, new object[] { handle });
+            
+            if (result == null)
+                throw new Exception($"Type {trueType}'s factory method returned a null object. This could mean improperly defined bindings");
+
+            return result;
+        }
+        
+        private static MethodInfo? GetRecordFactory(System.Type type)
```

Comment:
We do not want reflection during runtime.

Reply:
Perhaps this class should be removed completely?

Reply:
Currently necessary because we do the same for objects. Perhaps we can get rid of this via #397